DE GUZMAN, Adrian Gil

# How to find stuff in Roboflow?



1. **Project Id** and **Workspace Name**
  
  Look at the URL (Dataset view)
`https://app.roboflow.com/adeguzman/yolo-person-detection/generate/preprocessing`

  `adeguzman` - workspace name
  
  `yolo-person-detection` - project id

2. **API Key**

  Account Settings > Your Workspace (under workspaces) > API Keys

  `XvAG2BUPmPRgx9L24jYs` - API key

3. **Project ID** alternative

  Project > 3 dots beside project name > Copy Project ID

  `yolo-person-detection` - project ID


# Clone Ultralytics and Install Libraries

In [ ]:
# !git clone https://github.com/ultralytics/ultralytics
!pip install ultralytics
!pip install roboflow

# Import Libraries

In [2]:
from roboflow import Roboflow
from ultralytics import YOLO
import time

Creating new Ultralytics Settings v0.0.6 file ✅ 
View Ultralytics Settings with 'yolo settings' or at '/root/.config/Ultralytics/settings.json'
Update Settings with 'yolo settings key=value', i.e. 'yolo settings runs_dir=path/to/dir'. For help see https://docs.ultralytics.com/quickstart/#ultralytics-settings.


# Integrate Roboglow and Download Dataset

In [3]:
# Create Roboflow instance using API key
api_key = "XvAG2BUPmPRgx9L24jYs"
rf = Roboflow(api_key=api_key)

# Access Roboflow workspace "adeguzman"
# Look into project "yolo-person-detection"
workspace_name = "adeguzman"
project_id = "yolo-person-detection"
project = rf.workspace(workspace_name).project(project_id)

# Download version 1 of dataset using yolov8 format
version = 1
form = "yolov8"
dataset = project.version(version).download(form)

loading Roboflow workspace...
loading Roboflow project...



Extracting Dataset Version Zip to Yolo-Person-Detection-1 in yolov8:: 100%|██████████| 4780/4780 [00:01<00:00, 4003.06it/s]


# Train Yolov8n Model

In [4]:
# Load yolov8n.pt to initialize yolo model
model_v8n = YOLO('yolov8n.pt')

data_path = f'{dataset.location}/data.yaml'
epochs = 20
imgsz = 640

# Specify the project directory and project name
# to easily find the pt file (best.pt)
project_dir = 'saved_models'
model_name_v8n = 'yolov8n_model'

# Train the model
start_time_v11 = time.time()
# results_yolov8 = model_v8n.train(data=data_path, epochs=epochs, imgsz=imgsz)
results_yolov8 = model_v8n.train(data=data_path, epochs=epochs, imgsz=imgsz, project=project_dir, name=model_name_v8n)
end_time_v11 = time.time()

print(f"YOLOv8n training time: {end_time_v11 - start_time_v11:.2f} seconds.")

100%|██████████| 6.25M/6.25M [00:00<00:00, 103MB/s]


Ultralytics 8.3.129 🚀 Python-3.11.12 torch-2.6.0+cu124 CUDA:0 (Tesla T4, 15095MiB)
engine/trainer: agnostic_nms=False, amp=True, augment=False, auto_augment=randaugment, batch=16, bgr=0.0, box=7.5, cache=False, cfg=None, classes=None, close_mosaic=10, cls=0.5, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=/content/Yolo-Person-Detection-1/data.yaml, degrees=0.0, deterministic=True, device=None, dfl=1.5, dnn=False, dropout=0.0, dynamic=False, embed=None, epochs=20, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, half=False, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=640, int8=False, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.01, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.0, mode=train, model=yolov8n.pt, momentum=0.937, mosaic=1.0, multi_scale=False, name=yolov8n_model, nbs=64, nms=False, opset=None, optimize=False, optimizer=auto, overlap_mask=True, patience=100, perspective=0.0, plots=True,

100%|██████████| 755k/755k [00:00<00:00, 21.3MB/s]

Overriding model.yaml nc=80 with nc=1

                   from  n    params  module                                       arguments                     
  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]                 
  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]                
  2                  -1  1      7360  ultralytics.nn.modules.block.C2f             [32, 32, 1, True]             
  3                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                
  4                  -1  2     49664  ultralytics.nn.modules.block.C2f             [64, 64, 2, True]             
  5                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               
  6                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]           
  7                  -1  1    295424  ultralytics

Model summary: 129 layers, 3,011,043 parameters, 3,011,027 gradients, 8.2 GFLOPs

Transferred 319/355 items from pretrained weights
Freezing layer 'model.22.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks...


100%|██████████| 5.35M/5.35M [00:00<00:00, 116MB/s]


AMP: checks passed ✅
train: Fast image access ✅ (ping: 0.0±0.0 ms, read: 1687.3±676.4 MB/s, size: 85.5 KB)


train: Scanning /content/Yolo-Person-Detection-1/train/labels... 2076 images, 105 backgrounds, 0 corrupt: 100%|██████████| 2076/2076 [00:00<00:00, 2446.86it/s]

train: New cache created: /content/Yolo-Person-Detection-1/train/labels.cache


albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))
val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 763.0±858.4 MB/s, size: 101.8 KB)


val: Scanning /content/Yolo-Person-Detection-1/valid/labels... 204 images, 8 backgrounds, 0 corrupt: 100%|██████████| 204/204 [00:00<00:00, 1703.37it/s]

val: New cache created: /content/Yolo-Person-Detection-1/valid/labels.cache


Plotting labels to saved_models/yolov8n_model/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.002, momentum=0.9) with parameter groups 57 weight(decay=0.0), 64 weight(decay=0.0005), 63 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 2 dataloader workers
Logging results to saved_models/yolov8n_model
Starting training for 20 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       1/20      2.19G      1.453       1.87      1.355         68        640: 100%|██████████| 130/130 [00:39<00:00,  3.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:03<00:00,  1.81it/s]

                   all        204        552      0.562      0.554      0.583      0.293



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       2/20      2.59G      1.527      1.695      1.424         33        640: 100%|██████████| 130/130 [00:35<00:00,  3.69it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:02<00:00,  2.36it/s]

                   all        204        552      0.483      0.528      0.494      0.244



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       3/20      2.61G      1.591      1.704       1.49         55        640: 100%|██████████| 130/130 [00:35<00:00,  3.70it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:02<00:00,  2.45it/s]

                   all        204        552      0.503      0.487      0.457      0.218



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       4/20      2.63G      1.569      1.645      1.486         80        640: 100%|██████████| 130/130 [00:34<00:00,  3.77it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  3.51it/s]

                   all        204        552      0.665      0.551        0.6      0.327



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       5/20      2.64G      1.502      1.559      1.445         65        640: 100%|██████████| 130/130 [00:34<00:00,  3.72it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  3.66it/s]

                   all        204        552      0.554      0.625      0.595      0.296



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       6/20      2.66G      1.452      1.469      1.412         55        640: 100%|██████████| 130/130 [00:35<00:00,  3.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  3.62it/s]

                   all        204        552      0.622      0.599      0.596      0.319



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       7/20      2.68G      1.417       1.42      1.391         70        640: 100%|██████████| 130/130 [00:35<00:00,  3.65it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  3.65it/s]

                   all        204        552      0.663      0.665      0.685      0.376



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       8/20       2.7G      1.369      1.347      1.363         52        640: 100%|██████████| 130/130 [00:35<00:00,  3.63it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  3.76it/s]

                   all        204        552       0.67      0.689      0.704      0.388



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       9/20      2.71G      1.334      1.275      1.326         82        640: 100%|██████████| 130/130 [00:35<00:00,  3.69it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  3.67it/s]

                   all        204        552      0.675      0.682      0.717      0.421



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      10/20      2.73G      1.319       1.28      1.325         59        640: 100%|██████████| 130/130 [00:35<00:00,  3.61it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  3.83it/s]

                   all        204        552      0.676      0.674      0.694      0.397


Closing dataloader mosaic
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      11/20      2.75G      1.263      1.133        1.3         41        640: 100%|██████████| 130/130 [00:35<00:00,  3.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  3.79it/s]

                   all        204        552      0.708      0.715      0.759      0.437



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      12/20      2.76G      1.228      1.066      1.291         52        640: 100%|██████████| 130/130 [00:32<00:00,  3.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:02<00:00,  2.44it/s]

                   all        204        552      0.749      0.752      0.786      0.477



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      13/20      2.78G      1.166      1.018      1.239         29        640: 100%|██████████| 130/130 [00:32<00:00,  3.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  3.77it/s]

                   all        204        552      0.751      0.726      0.769      0.463



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      14/20       2.8G      1.138     0.9756      1.222         39        640: 100%|██████████| 130/130 [00:33<00:00,  3.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  3.79it/s]

                   all        204        552      0.753      0.759      0.808      0.511



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      15/20      2.81G      1.094     0.9066      1.191         28        640: 100%|██████████| 130/130 [00:33<00:00,  3.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  3.71it/s]

                   all        204        552      0.749      0.726      0.772      0.485



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      16/20      2.83G       1.04     0.8679      1.158         35        640: 100%|██████████| 130/130 [00:33<00:00,  3.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:02<00:00,  2.76it/s]

                   all        204        552      0.738      0.772      0.801      0.521



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      17/20      2.85G      1.042     0.8398      1.163         44        640: 100%|██████████| 130/130 [00:32<00:00,  3.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  4.07it/s]

                   all        204        552      0.727      0.799      0.815      0.545



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      18/20      2.87G     0.9746     0.7912      1.129         17        640: 100%|██████████| 130/130 [00:33<00:00,  3.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  3.99it/s]

                   all        204        552      0.778      0.766       0.83       0.56



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      19/20      2.88G     0.9682     0.7619      1.119         63        640: 100%|██████████| 130/130 [00:34<00:00,  3.81it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  3.95it/s]

                   all        204        552      0.751      0.775      0.804      0.543



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      20/20       2.9G     0.9195     0.7395      1.097         34        640: 100%|██████████| 130/130 [00:33<00:00,  3.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:02<00:00,  2.84it/s]

                   all        204        552      0.826      0.745      0.829      0.568



20 epochs completed in 0.207 hours.
Optimizer stripped from saved_models/yolov8n_model/weights/last.pt, 6.2MB
Optimizer stripped from saved_models/yolov8n_model/weights/best.pt, 6.2MB

Validating saved_models/yolov8n_model/weights/best.pt...
Ultralytics 8.3.129 🚀 Python-3.11.12 torch-2.6.0+cu124 CUDA:0 (Tesla T4, 15095MiB)
Model summary (fused): 72 layers, 3,005,843 parameters, 0 gradients, 8.1 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:03<00:00,  1.91it/s]


                   all        204        552      0.827      0.745       0.83      0.567
Speed: 0.3ms preprocess, 2.3ms inference, 0.0ms loss, 5.1ms postprocess per image
Results saved to saved_models/yolov8n_model
YOLOv8n training time: 775.01 seconds.


## Evaluate YOLOv8n model

In [17]:
# Validate the model through mAP results
metrics_v8n = model_v8n.val()
yolov8n_map = metrics_v8n.box.map
print(f"YOLOv8n mAP: {yolov8n_map:.4f}")

Ultralytics 8.3.129 🚀 Python-3.11.12 torch-2.6.0+cu124 CUDA:0 (Tesla T4, 15095MiB)
val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 2313.3±687.1 MB/s, size: 74.1 KB)


val: Scanning /content/Yolo-Person-Detection-1/valid/labels.cache... 204 images, 8 backgrounds, 0 corrupt: 100%|██████████| 204/204 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 13/13 [00:03<00:00,  3.51it/s]


                   all        204        552      0.827      0.745       0.83      0.568
Speed: 1.8ms preprocess, 5.4ms inference, 0.0ms loss, 2.0ms postprocess per image
Results saved to saved_models/yolov8n_model8
YOLOv8n mAP: 0.5682


# Train YOLOv11 Model

In [6]:
# Train YOLOv11
model_11n = YOLO('yolo11n.pt')

model_name_11n = 'yolo11n_model'

start_time_v11 = time.time()
results_v11 = model_11n.train(data=data_path, epochs=epochs, imgsz=imgsz, project=project_dir, name=model_name_11n)
end_time_v11 = time.time()

print(f"YOLOv11 training time: {end_time_v11 - start_time_v11:.2f} seconds.")

Ultralytics 8.3.129 🚀 Python-3.11.12 torch-2.6.0+cu124 CUDA:0 (Tesla T4, 15095MiB)
engine/trainer: agnostic_nms=False, amp=True, augment=False, auto_augment=randaugment, batch=16, bgr=0.0, box=7.5, cache=False, cfg=None, classes=None, close_mosaic=10, cls=0.5, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=/content/Yolo-Person-Detection-1/data.yaml, degrees=0.0, deterministic=True, device=None, dfl=1.5, dnn=False, dropout=0.0, dynamic=False, embed=None, epochs=20, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, half=False, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=640, int8=False, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.01, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.0, mode=train, model=yolo11n.pt, momentum=0.937, mosaic=1.0, multi_scale=False, name=yolo11n_model, nbs=64, nms=False, opset=None, optimize=False, optimizer=auto, overlap_mask=True, patience=100, perspective=0.0, plots=True,

train: Scanning /content/Yolo-Person-Detection-1/train/labels.cache... 2076 images, 105 backgrounds, 0 corrupt: 100%|██████████| 2076/2076 [00:00<?, ?it/s]

albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))


val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 844.0±806.3 MB/s, size: 101.8 KB)


val: Scanning /content/Yolo-Person-Detection-1/valid/labels.cache... 204 images, 8 backgrounds, 0 corrupt: 100%|██████████| 204/204 [00:00<?, ?it/s]


Plotting labels to saved_models/yolo11n_model/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.002, momentum=0.9) with parameter groups 81 weight(decay=0.0), 88 weight(decay=0.0005), 87 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 2 dataloader workers
Logging results to saved_models/yolo11n_model
Starting training for 20 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       1/20      2.49G      1.465      2.009      1.366         68        640: 100%|██████████| 130/130 [00:41<00:00,  3.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:02<00:00,  2.66it/s]


                   all        204        552      0.558      0.507      0.524      0.241

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       2/20      2.65G      1.556      1.785      1.442         33        640: 100%|██████████| 130/130 [00:37<00:00,  3.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:02<00:00,  3.09it/s]

                   all        204        552      0.497      0.478      0.461      0.202



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       3/20      2.65G      1.598      1.755      1.512         55        640: 100%|██████████| 130/130 [00:37<00:00,  3.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  3.90it/s]

                   all        204        552      0.461      0.493      0.433      0.182



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       4/20      2.65G      1.576      1.693      1.491         80        640: 100%|██████████| 130/130 [00:38<00:00,  3.39it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  3.94it/s]

                   all        204        552      0.596      0.524      0.557      0.281



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       5/20      2.65G      1.521      1.585       1.46         65        640: 100%|██████████| 130/130 [00:38<00:00,  3.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:02<00:00,  3.22it/s]

                   all        204        552      0.498      0.563      0.512      0.211



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       6/20      2.65G      1.463      1.509      1.425         55        640: 100%|██████████| 130/130 [00:39<00:00,  3.27it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:02<00:00,  3.46it/s]

                   all        204        552       0.67      0.551      0.608      0.334



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       7/20      2.65G      1.445      1.454      1.401         70        640: 100%|██████████| 130/130 [00:39<00:00,  3.28it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  4.12it/s]

                   all        204        552      0.652      0.585      0.603      0.316



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       8/20      2.65G      1.392      1.363      1.363         52        640: 100%|██████████| 130/130 [00:39<00:00,  3.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  4.21it/s]

                   all        204        552      0.681      0.601      0.656      0.375



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       9/20      2.65G      1.358      1.324      1.342         82        640: 100%|██████████| 130/130 [00:38<00:00,  3.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  3.59it/s]

                   all        204        552      0.714      0.681      0.717      0.402



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      10/20      2.65G      1.355      1.316      1.346         59        640: 100%|██████████| 130/130 [00:37<00:00,  3.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:03<00:00,  2.32it/s]

                   all        204        552      0.738      0.663      0.703      0.413


Closing dataloader mosaic
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      11/20      2.65G      1.266      1.152      1.301         41        640: 100%|██████████| 130/130 [00:37<00:00,  3.50it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:02<00:00,  2.47it/s]

                   all        204        552      0.705      0.664      0.706      0.399



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      12/20      2.65G      1.231      1.075      1.288         52        640: 100%|██████████| 130/130 [00:37<00:00,  3.49it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:02<00:00,  3.10it/s]

                   all        204        552      0.737      0.699      0.744      0.417



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      13/20      2.65G      1.181      1.026      1.246         29        640: 100%|██████████| 130/130 [00:36<00:00,  3.55it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  3.64it/s]

                   all        204        552      0.736      0.667      0.729      0.453



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      14/20      2.65G      1.148      0.987      1.232         39        640: 100%|██████████| 130/130 [00:35<00:00,  3.65it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:02<00:00,  2.92it/s]

                   all        204        552      0.723      0.745      0.775      0.486



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      15/20      2.65G      1.094     0.9404      1.195         28        640: 100%|██████████| 130/130 [00:36<00:00,  3.57it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:02<00:00,  3.42it/s]

                   all        204        552      0.702      0.726      0.743      0.466



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      16/20      2.65G      1.062     0.8944      1.172         35        640: 100%|██████████| 130/130 [00:35<00:00,  3.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:02<00:00,  3.07it/s]

                   all        204        552      0.725      0.746      0.769       0.51



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      17/20      2.65G      1.042     0.8619      1.164         44        640: 100%|██████████| 130/130 [00:35<00:00,  3.65it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:02<00:00,  2.84it/s]

                   all        204        552      0.714      0.774      0.782      0.514



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      18/20      2.65G     0.9859     0.8246       1.14         17        640: 100%|██████████| 130/130 [00:37<00:00,  3.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:02<00:00,  2.93it/s]

                   all        204        552      0.713      0.807      0.788      0.515



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      19/20      2.65G     0.9884     0.7983      1.132         63        640: 100%|██████████| 130/130 [00:38<00:00,  3.37it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:02<00:00,  3.23it/s]

                   all        204        552       0.74      0.777      0.795      0.535



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      20/20      2.65G     0.9343     0.7697      1.107         34        640: 100%|██████████| 130/130 [00:38<00:00,  3.36it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:01<00:00,  3.87it/s]

                   all        204        552      0.739      0.779      0.815      0.544



20 epochs completed in 0.225 hours.
Optimizer stripped from saved_models/yolo11n_model/weights/last.pt, 5.5MB
Optimizer stripped from saved_models/yolo11n_model/weights/best.pt, 5.5MB

Validating saved_models/yolo11n_model/weights/best.pt...
Ultralytics 8.3.129 🚀 Python-3.11.12 torch-2.6.0+cu124 CUDA:0 (Tesla T4, 15095MiB)
YOLO11n summary (fused): 100 layers, 2,582,347 parameters, 0 gradients, 6.3 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 7/7 [00:03<00:00,  1.84it/s]


                   all        204        552      0.739      0.779      0.815      0.544
Speed: 0.3ms preprocess, 3.4ms inference, 0.0ms loss, 4.9ms postprocess per image
Results saved to saved_models/yolo11n_model
YOLOv11 training time: 824.95 seconds.


## Evaluate YOLO11n Model

In [14]:
# Validate the model through mAP results
metrics_11n = model_11n.val()
yolo11n_map = metrics_11n.box.map
print(f"YOLO11n mAP: {yolo11n_map:.4f}")

Ultralytics 8.3.129 🚀 Python-3.11.12 torch-2.6.0+cu124 CUDA:0 (Tesla T4, 15095MiB)
val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 2242.7±518.5 MB/s, size: 67.8 KB)


val: Scanning /content/Yolo-Person-Detection-1/valid/labels.cache... 204 images, 8 backgrounds, 0 corrupt: 100%|██████████| 204/204 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 13/13 [00:03<00:00,  3.41it/s]


                   all        204        552       0.74      0.779      0.815      0.545
Speed: 2.5ms preprocess, 4.7ms inference, 0.0ms loss, 1.9ms postprocess per image
Results saved to saved_models/yolo11n_model4
YOLO11n mAP: 0.5454


# Conclusion

In [16]:
print(f"YOLOv8n mAP: {yolov8n_map:.4f}")
print(f"YOLO11n mAP: {yolo11n_map:.4f}")

YOLOv8n mAP: 0.5682
YOLO11n mAP: 0.5454


Interpretation Based on Geeks for Geeks Guide for Interpreting mAP
https://www.geeksforgeeks.org/mean-average-precision-map-in-computer-vision/

How to Interpret mAP Values?
- 0 to 1 (or 0% to 100%): The mAP score ranges from 0 to 1, where 1 indicates perfect precision and recall for all classes, and 0 indicates the worst performance.
- Closer to 1 (or 100%): Indicates a model that accurately detects and localizes objects with minimal false positives and false negatives. It reflects a well-performing model that can be reliably used in practical applications.
- Closer to 0: Indicates a model that struggles with object detection, producing many false positives and/or false negatives. It reflects a need for model improvement, better data, or more effective training.

Final Interpretation:

Since YOLOv8n mAP: 0.5682 > YOLO11n mAP: 0.5454, it indicates that YOLOv8n being the closer to 1 than YOLO11n is a more accurate model than the latter. After identifying the model with the better mAP, I selected its best.pt file, which contains the model's best-performing weights during training. This is preferred over last.pt, which only stores the weights from the final epoch. To easily locate the .pt files, we used the project parameter to define the name of the directory where training outputs are saved, and the name parameter to specify the subdirectory within that project. This setup organizes training logs, metrics, and model checkpoints in a clear and accessible structure.